astroid/brain/brain_typing.py

```diff
@@ -232,9 +232,7 @@ def _looks_like_typing_alias(node: Call) -> bool:
         and node.func.name == "_alias"
         and (
             # _alias function works also for builtins object such as list and dict
-            isinstance(node.args[0], Attribute)
-            or isinstance(node.args[0], Name)
-            and node.args[0].name != "type"
```

Comment:
Noticed that `type` was excluded as well. This was originally added with #921, I couldn't however find a good reason to keep it that way. It works as expected even for `type`. Could be that the `type` brain improved since then, idk.

I did run the acceptance test that caused headaches a while back. Worked without problems.
```
pytest -m acceptance -k "test_libmodule[typing.py]"
```